src/powersystem.jl

```diff
@@ -60,15 +62,194 @@ struct PowerNetwork
+            length(pf.pv), length(pf.pq))
+    println("\tGenerators: ", pf.ngen, ".")
+    println("\tLoads: ", pf.nload, ".")
+
+    # Print system status
+    @printf("\t==============================================\n")
+    @printf("\tBUS \t TYPE \t VMAG \t VANG \t P \t Q\n")
+    @printf("\t==============================================\n")
+    
+    for i=1:pf.nbus
+        type = pf.bustype[i]
+        vmag = abs(pf.V[i])
+        vang = angle(pf.V[i])*(360.0/pi)
+        pinj = real(pf.Sbus[i])
+        qinj = imag(pf.Sbus[i])
+        @printf("\t%i \t  %d \t %1.3f\t%3.2f\t%3.3f\t%3.3f\n", i,
+                type, vmag, vang, pinj, pinj)
     end
+
 end
 
 
+
+
+"""
+    assemble_vecs(PowerNetwork)
+
+Returns vectors x, u, p from network variables (VMAG, VANG, P and Q)
+and bus type info.
+
+
+Vector x is the variable vector, consisting on:
+    - VMAG, VANG for buses of type PQ (type 1)
+    - VANG for buses of type PV (type 2)
+These variables are determined by the physics of the network.
+
+Vector u is the control vector, consisting on:
+    - VMAG, P for buses of type PV (type 1)
+    - VM for buses of type SLACK (type 3)
+These variables are controlled by the grid operator.
+
+Vector p is the parameter vector, consisting on:
+    - VA for buses of type SLACK (type 3)
+    - P, Q for buses of type PQ (type 1)
+These parameters are fixed through the problem.
+
+In addition, variables:
+    - Q for buses of type PV (type 2)
+    - P, Q for buses of type SLACK (type 3)
+are explicitly obtained from the previous variables, and hence
+not needed in the computation.
+"""
+function assemble_vecs(pf::PowerNetwork)
+ 
+    nref = length(pf.ref)
+    npv = length(pf.pv)
+    npq = length(pf.pq)
+    println(nref, npv, npq)
+
+
+    # build vector x
+    dimension = 2*npq + npv
+    x = zeros(dimension, 1)
+    k = 1
+
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 1 #PQ
+            x[k] = abs(pf.V[bus])
+            x[k + 1] = angle(pf.V[bus])
+            k = k + 2
+        elseif pf.bustype[bus] == 2 #PV
+            x[k] = angle(pf.V[bus])
+            k = k + 1
+        end
+    end
+
+    # build vector u
+    dimension = 2*npv + nref
+    u = zeros(dimension, 1)
+    k = 1
+
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 2 #PV
+            u[k] = abs(pf.V[bus])
+            u[k + 1] = real(pf.Sbus[bus])
+            k = k + 2
+        elseif pf.bustype[bus] == 3 # slack
+            u[k] = abs(pf.V[bus])
+            k = k + 1
+        end
+    end
+
+    # build vector p
+    dimension = nref + 2*npq
+    p = zeros(dimension, 1)
+    k = 1
+    
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 3 #slack
+            p[k] = abs(pf.V[bus])
+            k = k + 1
+        elseif pf.bustype[bus] == 1 # PQ
+            p[k] = real(pf.Sbus[bus])
+            p[k + 1] = imag(pf.Sbus[bus])
+            k = k + 2
+        end
+    end
+
+    return x, u, p
+
+end
+
+
+"""
+    xup_to_vs(PowerNetwork, x, u, p)
```

Comment:
I think we should also think about the naming for this function :)

Reply:
@frapac I agree. Is a terrible name. It was late at night...

Reply:
To be fairly honest, I am bad at naming even during the day ^^